packages/playground/src/app/infinite-query-page/posts.service.ts

```diff
@@ -0,0 +1,37 @@
+import { HttpClient } from '@angular/common/http';
+import { inject, Injectable } from '@angular/core';
+import { InfiniteQueryProvider } from '@ngneat/ng-query';
+import { tap } from 'rxjs';
+
+interface Post {
+  id: number;
+  userId: string;
+  title: string;
+  body: string;
+}
+
+@Injectable({ providedIn: 'root' })
+export class PostsService {
+  private http = inject(HttpClient);
+  private useInfiniteQuery = inject(InfiniteQueryProvider);
+
+  getPosts() {
+    return this.useInfiniteQuery<Post[]>(
```

Comment:
Can we please mimic the same example from the offical docs?

https://tanstack.com/query/v4/docs/examples/react/load-more-infinite-scroll

Reply:
tried my best. Please have a look